drivers/include/disp_dev.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2020 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_disp_dev Display device generic API
+ * @ingroup     sys
+ * @brief       Define the generic API of a display device
+ * @{
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef DISP_DEV_H
+#define DISP_DEV_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdbool.h>
+#include <stdint.h>
+
+/**
+ * @brief   Forward declaration for display device struct
+ */
+typedef struct disp_dev disp_dev_t;
+
+/**
+ * @brief   Generic type for a display driver
+ */
+typedef struct {
+    /**
+     * @brief   Map an area to display on the device
+     *
+     * @param[in] dev   Pointer to the display device
+     * @param[in] x1    Left coordinate
+     * @param[in] x2    Right coordinate
+     * @param[in] y1    Top coordinate
+     * @param[in] y2    Bottom coordinate
+     * @param[in] color Array of color to map to the display
+     */
+    void (*map)(disp_dev_t *dev,
```

Comment:
@bergzand @aabadie Is there any advantage in having `x1, x2, y1, y2, color[]` vs. `x, y, ptr, numpixels` or `x, y, ptr, bytes` or  `x, y, width, height, ptr`)?

I see that `numpixels` might lead to users putting the number of bytes there, which on 16bit colors would lead to nasty bugs. `bytes` would only lead to (smaller) garbage output.

`uint16_t*` might lead to alignment bugs, as e.g., blobs might get included as `const char *`.

I'm asking because I find `x1, x2, y1, y2` clumsy to work with.

Reply:
I'm just realizing that it must be `x1, x2, y1, y2, ptr` or `x, y, width, height, ptr` in order to support arbitrary rectangles.